CONFIGURATION.md

```diff
@@ -13,7 +13,7 @@ receive.message.max.bytes                |  *  | 1000 .. 2147483647 |     100000
 max.in.flight.requests.per.connection    |  *  | 1 .. 1000000    |       1000000 | low        | Maximum number of in-flight requests per broker connection. This is a generic property applied to all broker communication, however it is primarily relevant to produce requests. In particular, note that other mechanisms limit the number of outstanding consumer fetch request per broker to one. <br>*Type: integer*
 max.in.flight                            |  *  | 1 .. 1000000    |       1000000 | low        | Alias for `max.in.flight.requests.per.connection`: Maximum number of in-flight requests per broker connection. This is a generic property applied to all broker communication, however it is primarily relevant to produce requests. In particular, note that other mechanisms limit the number of outstanding consumer fetch request per broker to one. <br>*Type: integer*
 metadata.request.timeout.ms              |  *  | 10 .. 900000    |         60000 | low        | Non-topic request timeout in milliseconds. This is for metadata requests, etc. <br>*Type: integer*
-topic.metadata.refresh.interval.ms       |  *  | -1 .. 3600000   |        300000 | low        | Topic metadata refresh interval in milliseconds. The metadata is automatically refreshed on error and connect. Use -1 to disable the intervalled refresh. <br>*Type: integer*
+topic.metadata.refresh.interval.ms       |  *  | -1 .. 3600000   |        300000 | low        | Topic metadata refresh interval in milliseconds. The metadata is automatically refreshed on error and connect. Use -1 to disable the intervalled refresh. If there are no locally referenced topics (no topic objects created, no messages produced, no subscription or no assignment) then only the broker list will be refreshed every interval but no more often than every 10s. <br>*Type: integer*
```

Comment:
This description caused me to pause. Initially, because I felt it was too verbose, but on reflection, I think it's because no reason for the refresh was stated. Here's an alternative to consider (the first part somewhat inspired by the java docs). I left it in, but use of the word 'error' is debatable I think since this refers to re-tryable error most of the time I think - that the user doesn't see - and so is not an error from the users perspective (or at best the word is ambiguous from the users perspective).

> "The period of time in milliseconds after which a metadata refresh is forced in order to proactively discover any new brokers or partitions. Metadata is automatically refreshed on error and connect. Use -1 to disable the intervalled refresh. If there are no locally referenced topics (no topic objects created, no messages produced, no subscription or no assignment) then only the broker list will be refreshed every interval but no more often than every 10s."